sled-storage/src/manager.rs

```diff
@@ -933,17 +934,36 @@ impl StorageManager {
         log: &Logger,
         config: &DatasetsConfig,
     ) -> DatasetsManagementResult {
-        let mut status = vec![];
-        for dataset in config.datasets.values() {
-            status.push(self.dataset_ensure_internal(log, dataset).await);
-        }
+        // Gather properties about these datasets, if they exist.
+        //
+        // This pre-fetching lets us avoid individually querying them later.
+        let old_datasets = Zfs::get_dataset_properties(
```

Comment:
This query, as well as the logic below, is the primary optimization made by this PR.

We call `zfs get` once for all datasets we care about, and if no changes are necessary, that's the only process created.